### PR TITLE
Cardano owner support

### PR DESCRIPTION
**Pull request by refi93**
WIP PR to showcase work done so far on cardano owner support

### CODE REVIEW

File: src/js/core/methods/helpers/cardanoCertificate.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    ]);
+
+    switch (relay.type) {
+        case POOL_RELAY_TYPE.SingleHostIp: {
+            validateParams(relay, [
+                { name: 'ipv4Address', type: 'string', obligatory: !relay.ipv6Address },
+                { name: 'ipv6Address', type: 'string', obligatory: !relay.ipv4Address },
+                { name: 'port', type: 'number', obligatory: true },
+            ]);
+        }
```
Comment (gabrielKerekes) [inline src/js/core/methods/helpers/cardanoCertificate.js:57]: Aren't there missing `break;`s here?

File: tests/__fixtures__/cardanoSignTransaction.js [hunk 2]
```diff
@@ -123,7 +201,7 @@ export default {
         mnemonic: 'mnemonic_all',
     },
     tests: [
-        {
+        /*{
```
Comment (gabrielKerekes) [inline tests/__fixtures__/cardanoSignTransaction.js]: Forgotten comment?

File: src/js/types/trezor/protobuf.js [hunk 3]
```diff
@@ -602,19 +602,51 @@ export type CardanoSignedTx = {
...
+    port?: number;
+}
+
+export type CardanoPoolMetadata = {
+    url: string;
+    hash: string;
+}
+
+export type CardanoPoolParameters = {
+    vrf_key_hash: string;
```
Comment (gabrielKerekes) [inline src/js/types/trezor/protobuf.js:634]: Isn't pool missing here?

**Review by gabrielKerekes — CHANGES_REQUESTED** [review]
I think I found some errors, however I wasn't able to run the tests, so please take a look at the errors.

File: src/js/core/methods/helpers/cardanoCertificate.js (hunk 1 above)
Comment (refi93) [reply src/js/core/methods/helpers/cardanoCertificate.js:57]: yes, I had it already locally fixed but forgot to push :facepalm:  , sorry

**Review by gabrielKerekes — APPROVED** [review]

File: src/js/core/methods/helpers/cardanoCertificate.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+const validatePoolRelay = (relay: CardanoPoolRelay) => {
+    validateParams(relay, [
+        { name: 'type', type: 'number', obligatory: true },
+    ]);
+
+    if (relay.type === POOL_RELAY_TYPE.SingleHostIp) {
+        validateParams(relay, [
+            { name: 'ipv4Address', type: 'string', obligatory: !relay.ipv6Address },
+            { name: 'ipv6Address', type: 'string', obligatory: !relay.ipv4Address },
```
Comment (janmazak) [inline src/js/core/methods/helpers/cardanoCertificate.js]: Ledger is not validating this... it is totally reasonable to expect at least one form of IP address, but does the  absence of the ip address make the tx invalid?

File: src/js/core/methods/helpers/cardanoCertificate.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            { name: 'port', type: 'number', obligatory: true },
+        ]);
+    } else if (relay.type === POOL_RELAY_TYPE.SingleHostName) {
+        validateParams(relay, [
+            { name: 'hostName', type: 'string', obligatory: true },
+            { name: 'port', type: 'number', obligatory: true },
```
Comment (janmazak) [inline src/js/core/methods/helpers/cardanoCertificate.js:70]: CDDL suggests that port is not obligatory.

**Comment by refi93** [comment]
already merged in parent repo

**Pull request closed without merging**